src/hubmap_translator.py

```diff
@@ -57,9 +57,9 @@ class Translator(TranslatorInterface):
     skip_comparision = False
     failed_entity_api_calls = []
     failed_entity_ids = []
+    _ontology_api_dict = None
 
-
-    def __init__(self, indices, app_client_id, app_client_secret, token):
+    def __init__(self, indices, app_client_id, app_client_secret, token, ontology_api_dict:dict=None):
```

Comment:
@kburke I understand you made the application context configurable, there's really no need to do that, it'll be a fixed value `application_context=HUBMAP`. This is how the UBKG ontology-api hardcoded internally for HuBMAP and SenNet.

Passing in a single URL base is sufficient and easier.

Reply:
I was passing around the application_context, which I understand must be a part of the ontology-api URL for the first two reasons that follow:
- Initially I was trying to make this work as a replacement **_inside_** search-adaptor's [src/translator/tranlation_helper_functions.py](https://github.com/dbmi-pitt/search-adaptor/blob/main/src/translator/tranlation_helper_functions.py), which meant SN would be using the same code.  If you affirm me "pulling this up" into search-api, I can inline the `application_context` rather than read it from configuration.
- But if I inline the code, I will change [this line](https://github.com/hubmapconsortium/search-api/blob/a72bd216019916d4e71e0ceb0ea3bc5ec3ef9154/src/hubmap_translator.py#L1186) to be
`target_url = f"{ontology_api_url}/organs/by-code?application_context=HUBMAP"`, right?  And that will mean SN needs to notice it there and set it, rather than configuration.  Just want to confirm that is preferred for one-time, immutable settings...
- And is the URL any less immutable?  Should I inline that, to really simplify?

Reply:
@kburke I appreciate the thinking you putting into the work for compatibility considerations. In case I didn't fully understand you or maybe you didn't understand me, let's sort this out:
- SenNet doesn't use this `get_type_description()` from the helper functions, so it's totally fine the way you did by removing it: https://github.com/dbmi-pitt/search-adaptor/pull/50/commits/7b8e9121142789e6143ec78f42248340b75f510b
- Yes, we'll be using `target_url = f"{ontology_api_url}/organs/by-code?application_context=HUBMAP"` in the HuBMAP search-api code, and SenNet should not care about this. Consider SenNet search-api repo a totally different application.
- I treated the URL path as a constant in entity-api: https://github.com/hubmapconsortium/entity-api/blob/main/src/schema/schema_constants.py#L18 to minimize the actual code changes. In search-api, you can also define it at the top to assist the future change (unlikely): https://github.com/hubmapconsortium/search-api/blob/karlburke/removeYAMLuseUBKG/src/hubmap_translator.py#L49-L51

Reply:
Thanks, @yuanzhou , I understand and will get it in that way.

Reply:
Committed with these constants embedded where Joe suggests above, inside `hubmap_translator.py`.
This removes the need to commit `app.cfg.example`, as request at the top

Reply:
@kburke we need this base URL configurable in `app.cfg.example`

Reply:
@yuanzhou Returned the base URL to the app.cfg.example, left endpoint URL constant where I understand the 3rd comment of this chain suggests it be placed.